criu/arch/x86/restorer.c

```diff
@@ -2,6 +2,7 @@
 #include <unistd.h>
 
 #include "types.h"
+#include "flog.h"
```

Comment:
Why is this header needed here?

Reply:
Due to the presence of logging macros in this file and all the header files it includes, the absence of this header throws me an error: implicit declaration of function ‘flog_encode’. To avoid this error I had added this header here.

Reply:
OK, so you took my words about not to include headers in headers too literally :) Here's how it's usually done:

- If you have a lot of .c files, that need something from some header, then you include this header into all of them. Even if there's a shortcut -- by including this header into some other one, which is in turn already present in all those .c files -- you still go and include the header into .c files.

- If you have a .h file, that needs something from another header, then you include header into header. There's another solution to this -- you go and include this another header in all the .c files, that have the 1st header, but it's not the way to go either.

Regarding this place -- the criu/arch/x86/restorer.c does not need the flog.h by itself. The flog.h turns out to be needed by restorer.c because restorer.c includes log.h which needs flog. This is the 1st item in my example, so include flog.h into log.h and that's it.

Reply:
I have already included flog.h in the criu/include/log.h. Even after the addition of flog.h in log.h, during the compile time I get this error, if I do not include this header here.